src/components/admin-route/admin-route.tsx

```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { Navigate, Outlet } from "react-router-dom";
+
+import { AppRoute } from "~/libs/constants/index.ts";
+import { useAppSelector } from "~/redux/hooks.ts";
+import { RootState } from "~/redux/store.ts";
+
+const AdminRoute: React.FC = () => {
+	const role = "admin";
```

Comment:
move out of component. I would even move it to constant and create object with all roles